### PR TITLE
add basic commenting to pages

### PR DESCRIPTION
**Pull request by Rizziepit**
Implements comment posting and listing for the `content` view

### CODE REVIEW

**Comment by Rizziepit** [comment]
@miltontony this could use an early review. Some things worth noting:
* It uses [deform](http://deform.readthedocs.org/en/latest/) for form handling.
* Comments won't show up unless the client is configured and the stream has been enabled.
* The comment form doesn't implement anti-spam things because only authenticated users can comment (for now).

I'm going to put tests, flagging and jinja templates in their own branches that will merge back into this branch - this branch is getting a bit big.

File: cms/views/cms_views.py [hunk 1]
```diff
@@ -242,13 +295,47 @@ def content(self):
...
             'linked_pages': linked_pages,
             'primary_category': category,
             'content': markdown(page.content),
             'description': markdown(page.description),
         }
+        context.update(self.get_comment_context(page))
+        return context
+
+    @view_config(route_name='comments',
+                 renderer='cms:templates/comments/comment_page.pt')
```
Comment (miltontony) [inline cms/views/cms_views.py]: I think you shouldn't be bothering with chameleon templates. I think you should just implement these using Jinja2

Comment (Rizziepit) [inline cms/views/cms_views.py]: That means existing sites will have to switch to Jinja templates if they want commenting. Is that okay?

Comment (miltontony) [inline cms/views/cms_views.py]: So we already have Jinja2 templates in this codebase and we'll be moving over to springboard _very soon_ so I'd rather not have any new templates in the old way (it just means more work later). And we won't be enabling commenting on the old templates (I refuse ;) )

**Comment by miltontony** [comment]
looks good so far

File: cms/views/cms_views.py (hunk 1 above)
Comment (Rizziepit) [inline cms/views/cms_views.py]: Fair enough. I'm going to implement the Jinja templates on another branch to make review easy and then merge it into this branch.

Comment (miltontony) [inline cms/views/cms_views.py]: :ok_hand:

**Comment by Rizziepit** [comment]
@miltontony I'm going to merge this branch now. All the new code here has been reviewed on other branches.

**Comment by miltontony** [comment]
@Rizziepit probably worth doing a `git pull origin develop` to sync it with develop?

**Comment by miltontony** [comment]
:+1:

**Pull request merged**